system_tests/datastore.py

```diff
@@ -271,7 +287,13 @@ def test_projection_query(self):
         self.assertEqual(catelyn_stark_dict,
                          {'name': 'Catelyn', 'family': 'Stark'})
 
-        catelyn_tully_entity = entities[3]
+        if EMULATOR_DATASET is None:
```

Comment:
@pcostell This was surprising to me.

1. Why have we ever been able to rely on hardcoding the index returned from a projection query?
1. Why is the order different on the emulator than the backend?